Dntc.Common/Syntax/Statements/Generators/LineInfoStatementGenerator.cs

```diff
@@ -0,0 +1,71 @@
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+namespace Dntc.Common.Syntax.Statements.Generators;
+
+public enum LineInfoMode
+{
+    Directives,
+    Comments,
+    None,
+}
+
+public class LineInfoStatementGenerator(LineInfoMode mode) : IStatementGenerator
+{
+    private EndLineStatementSet? _movedStatementSet;
+    
+    public IEnumerable<CStatementSet> Before(List<CStatementSet> statements, MethodDefinition definition,
+        Instruction methodInstruction)
+    {
+        if (mode != LineInfoMode.None)
+        {
+            var sequencePoint = CecilUtils.GetSequencePoint(definition, methodInstruction);
+
+            var lastLineStatement = statements.OfType<LineStatementSet>().LastOrDefault();
+
+            if (lastLineStatement != null && lastLineStatement.SequencePoint == sequencePoint)
+            {
+                var lastEndLineStatement = statements.OfType<EndLineStatementSet>().LastOrDefault();
+                if (lastEndLineStatement != null)
+                {
+                    statements.Remove(lastEndLineStatement);
```

Comment:
Am I correct that this is handling the case where multiple line statement generators are in play for the same section, so we merge their line statements together?

And what's the reason for the last end line statement from line 43 to be assigned to `_movedstatementSet` but not this one?

Reply:
yes thats correct, merging where sequence points are the same.
Iv added some comments to make this more understandable.

The moveStatementSet is for handling hidden sequence points, they essentially become part of the same line, so we just keep moving the endLineStatement until there is a non hidden sequence point.